### PR TITLE
Remove host-creation endpoint, tests, etc.

### PR DESCRIPTION
**Pull request by kruai**
Addresses this Jira: https://issues.redhat.com/browse/RHCLOUD-10520
Removes the host-creation API endpoint, plus tests, and code that is no longer used as a result.

### CODE REVIEW

**Review by thearifismail — APPROVED** [review]
LGTM

File: tests/test_system_profile.py [hunk 1]
```diff
@@ -106,3 +112,57 @@ def test_get_system_profile_sap_sids_with_RBAC_bypassed_as_system(
     response_status, response_data = api_get(url, identity_type="System")
 
     assert_response_status(response_status, 200)
+
+
+def test_get_system_profile_RBAC_allowed(mocker, subtests, api_get, db_create_host, enable_rbac):
```
Comment (ryandillinfelton) [inline tests/test_system_profile.py:117]: Why is adding these test part of this PR?

**Review by ryandillinfelton — CHANGES_REQUESTED** [review]
I don't think the test_system_profile.py test additions should be part of this PR

File: tests/test_system_profile.py (hunk 1 above)
Comment (kruai) [reply tests/test_system_profile.py:117]: They're not being added - they're being moved from `test_api_hosts_create.py` (they should have been in `test_system_profile.py` to begin with imo)

Comment (ryandillinfelton) [reply tests/test_system_profile.py:117]: Oh I see, they were in the wring file before. Switching to approve :+1:

**Review by ryandillinfelton — APPROVED** [review]

**Review by tahmidefaz — APPROVED** [review]

**Comment by Glutexo** [comment]
I am glad this is no longer part of #735.

**Pull request merged**